GooglePlacesAutocomplete.js

```diff
@@ -628,6 +643,10 @@ export const GooglePlacesAutocomplete = forwardRef((props, ref) => {
       setRequestHeaders(request, getRequestHeaders(props.requestUrl));
 
       if (props.isNewPlacesAPI) {
+        request.setRequestHeader(
+          'content-type',
```

Comment:
On Android native devices, the API request fails with the response message stating that it needs `content-type` header. Adding the header resolves the issue with Android native devices.

Reply:
Can you help me understand why specifically this happens on android and if this is the only fix? As you stated in your App PR, adding this header causes CORS errors - I noticed this on dev too.

Reply:
The response (as shared in screenshot below) precisely pointed out this reason and so didn’t dig deeper. But your question on why specifically this happens on Android is interesting. I will take a closer look at this.

![Screenshot 2024-10-09 at 1 23 01 PM](https://github.com/user-attachments/assets/ece0a860-882d-48a4-81b5-c2c0762a72c2)

Reply:
Thanks yeah please do, it is very strange. AFAIK that response is not coming from the server

Reply:
> Thanks yeah please do, it is very strange. AFAIK that response is not coming from the server

@rafecolton That's correct. The response comes from android native implementation as referenced in the screenshot below. Since we have stringified the request body of our POST request, android native implementation expects `content-type` to be sent. The only way to programmatically avoid this response is to avoid JSON.stringify but it does not seem to be a viable option for us. 
So, It looks like we have to address the issue in `desktop` platform along with the `content-type` and we are discussing this [here](https://github.com/Expensify/App/pull/47085#issuecomment-2398570019)

![Screenshot 2024-10-29 at 6 33 48 AM](https://github.com/user-attachments/assets/e1a34f43-2c7f-4a73-a1fd-0a87fe7ff412)